test/checkpoint.test.js

```diff
@@ -24,5 +25,36 @@ describe('Checkpoint', function() {
         }
       ], done);
     });
+
+    it('Should be no race condition when calling in parallel', function(done) {
+      async.parallel([
+        function(next) { Checkpoint.current(next); },
+        function(next) { Checkpoint.current(next); }
+      ], function(err, list) {
+        if (err) return done(err);
+        Checkpoint.find(function(err, data) {
+          if (err) return done(err);
+          expect(data).to.have.length(1);
+          done();
+        });
+      });
+    });
+
+    it('Checkpoint.current() for non existing checkpoint should initialize checkpoint', function(done) {
+      Checkpoint.current(function(err, seq) {
+        expect(seq).to.equal(1);
+        done(err);
+      });
+    });
+
+    it('bumpLastSeq() works when singleton instance does not exists yet', function(done) {
+      Checkpoint.bumpLastSeq(function(err, cp) {
+        // We expect `seq` to be 2 since `checkpoint` does not exist and
+        // `bumpLastSeq` for the first time not only initializes it to one,
+        // but also increments the initialized value by one.
+        expect(cp.seq).to.equal(2);
```

Comment:
Not all connectors report whether `findOrCreate` creates a new instance or not. Let's keep the current implementation as it simpler. I think the extra cost caused by the more complex version is not worth the little gain it brings.